### PR TITLE
fix large file upload issue #88

### PR DESCRIPTION
**Pull request by xjin-splunk**
Implemented the feature of uploading local files that are larger than 32MB mentioned at [here](https://github.com/crazy-max/ghaction-virustotal/issues/88). It was accomplished by requesting the special large file upload link suggested [here](https://developers.virustotal.com/reference/files-upload-url). Could you help review my code? Thank you.

### CODE REVIEW

File: src/virustotal.ts [hunk 1]
```diff
@@ -20,10 +20,11 @@ export interface Asset {
 
 export class VirusTotal {
   private instance: AxiosInstance;
+  private largeURL = '';
 
   constructor(apiKey: string | undefined) {
     this.instance = axios.create({
-      baseURL: 'https://www.virustotal.com/api/v3',
+      baseURL: '',
```
Comment (crazy-max) [inline src/virustotal.ts:27]: `baseURL` should not be changed here.

File: src/virustotal.ts [hunk 2]
```diff
@@ -32,6 +33,24 @@ export class VirusTotal {
     });
   }
 
+  async getURL(apiKey: string | undefined): Promise<string> {
```
Comment (crazy-max) [inline src/virustotal.ts:36]: Should be private and used in `files` method.

File: src/virustotal.ts [hunk 3]
```diff
@@ -32,6 +33,24 @@ export class VirusTotal {
...
+    });
+
+    return await instance.get('https://www.virustotal.com/api/v3/files/upload_url').then(res => {
+      return res.data.data;
+    });
+  }
+
+  async getLargeFileURL(apiKey: string | undefined) {
+    this.largeURL = await this.getURL(apiKey);
+  }
```
Comment (crazy-max) [inline src/virustotal.ts:52]: Not needed if `getURL` directly called in `files`

File: src/virustotal.ts [hunk 4]
```diff
@@ -42,7 +61,7 @@ export class VirusTotal {
     });
 
     return this.instance
-      .post('/files', fd.getBuffer(), {
+      .post(this.largeURL, fd.getBuffer(), {
```
Comment (crazy-max) [inline src/virustotal.ts:64]: As stipulated in their doc "Each upload URL can be used only once." so we should call `getURL` instead.

We should also get a URL for uploading large files only if file size > 32MB, not every time as we are going to quickly hit the rate limit.

File: src/virustotal.ts [hunk 5]
```diff
@@ -69,7 +88,7 @@ export class VirusTotal {
     fd.append('path', itemPath);
 
     return this.instance
-      .post('/monitor/items', fd.getBuffer(), {
+      .post('https://www.virustotal.com/api/v3/monitor/items', fd.getBuffer(), {
```
Comment (crazy-max) [inline src/virustotal.ts:91]: See my previous comment about reverting baseURL change.

**Review by crazy-max — CHANGES_REQUESTED** [review]

**Comment by xjin-splunk** [comment]
Hello, thank you for the feedbacks. I have updated the changes. Could you help take other look?

File: src/virustotal.ts [hunk 6]
```diff
@@ -32,7 +34,21 @@ export class VirusTotal {
...
+      baseURL: '',
+      headers: {
+        Accept: 'application/json',
+        'x-apikey': apiKey ?? ''
+      }
+    });
+
+    return await instance.get('https://www.virustotal.com/api/v3/files/upload_url').then(res => {
+      return res.data.data;
+    });
```
Comment (crazy-max) [inline src/virustotal.ts:48]: Just use `return this.instance.post('/files/upload_url')`. No need to create another instance, the one from the constructor is fine.

Comment (crazy-max) [reply src/virustotal.ts:48]: Actually I think we could just move this call in `files` method and remove the `getURL` method.

**Review by crazy-max — CHANGES_REQUESTED** [review]
Some minor changes. I you can, also add tests for this but not a blocker, can be done in a follow-up.

File: src/virustotal.ts (hunk 6 above)
Comment (xjin-splunk) [reply src/virustotal.ts]: Hi, this one line of code solution is indeed more elegant. I have made the change, though the style checker requires it to be in multiple lines.

**Review by crazy-max — APPROVED** [review]
LGTM thanks! ci issue is unrelated and will be fixed in a follow-up.

**Comment by xjin-splunk** [comment]
Thank you or merging my PR! May I ask when will there be a new release/tag that includes this update?

**Pull request merged**